aerospike-core/src/client.rs

```diff
@@ -88,52 +87,48 @@ impl Client {
     ///
     /// Using an environment variable to set the list of seed hosts.
     ///
-    /// ```rust
+    /// ```rust,edition2018
     /// use aerospike::{Client, ClientPolicy};
     ///
     /// let hosts = std::env::var("AEROSPIKE_HOSTS").unwrap();
-    /// let client = Client::new(&ClientPolicy::default(), &hosts).unwrap();
+    /// let client = Client::new(&ClientPolicy::default(), &hosts).await.unwrap();
     /// ```
-    pub fn new(policy: &ClientPolicy, hosts: &dyn ToHosts) -> Result<Self> {
+    pub async fn new(policy: &ClientPolicy, hosts: &(dyn ToHosts + Send + Sync)) -> Result<Self> {
```

Comment:
I am not sure why `hosts` needs to be `&dyn`.

`hosts: impl AsRef<ToHosts> + Send + Sync` might be better.

Reply:
+1. The interface implementation seems to be better to me.